pytest_mojo/plugin.py

```diff
@@ -83,11 +90,13 @@ def collect(self):
 
 class MojoTestItem(Item):
     def __init__(self, *, name: str, parent, spec: dict[str, Any], **kwargs):
-        super().__init__(name, parent, **kwargs)
+        super().__init__(name.removeprefix(TEST_ITEM_PREFIX), parent, **kwargs)
         self.spec = spec
 
     def runtest(self):
-        if self.spec["code"] != 0:
+        if self.spec.get("code") or any(
+            item.startswith(TEST_FAILED_PREFIX) for item in self.spec["stdout"]
```

Comment:
I think this logic needs to move into `MojoTestFile.collect()`. The reason is the running of mojo and parsing of results is actually happening in `MojoTestFile`. The `MojoTestItem` is just a report formatter, more or less. The general approach is explained better in this article, https://pytest-c-testrunner.readthedocs.io/ although it is apparently using an older version of pytest so the code won't exactly align.

Reply:
https://pytest-c-testrunner.readthedocs.io/conftest.html#conftest

> """The test has already been run. We just evaluate the result."""

Reply:
Actually, never mind, I think that logic is ok where it is. Looking for TEST_FAILED_PREFIX makes sense to do in the "evaluate the result" section of of the code.